nuxeo-common/src/test/java/org/nuxeo/common/utils/TestTextTemplate.java

```diff
@@ -157,4 +158,32 @@ public void testEscapeVariable() {
         assertEquals("{$$CZjbsiX748UF583qkbinsQ==}", templates.processText("{$$CZjbsiX748UF583qkbinsQ==}"));
     }
 
+    @Test
+    public void testFreetemplateFileNameValidity() throws FileNotFoundException, IOException, TemplateException {
+        final TextTemplate textTemplate = new TextTemplate();
+        textTemplate.setFreemarkerParsingExtensions("nxftl");
+
+        final File testResouceDirectory = new File(getClass().getClassLoader().getResource("test-nxftl").getPath());
+        final File tmpdir = new File(System.getProperty("java.io.tmpdir"));
```

Comment:
Use `File.createTempFile` to create temporary files or directories, and remove it in a `finally` block. Pattern to create an empty temporary directory is:

    File tmpDir = File.createTempFile();
    tmpDir.delete();

If you can (if you don't have to use legacy File APIs) use `Path` from `java.nio.file` instead, and `java.nio.file.Files` to create temporary stuff or other path operations.

Reply:
Ok, will do.